dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/NeuralNetwork.java

```diff
@@ -187,10 +193,15 @@ public void train(final INDArray input, final int label) {
                                     final INDArray label) {
     final int lastLayerIndex = layers.length - 1;
 
-    // The first element of activations is input data.
+    // The first element of activations is the input data.
     // So, (i + 1)-th element of activations refers to the activation of i-th layer.
-    final INDArray error = layers[lastLayerIndex].backPropagate(activations[lastLayerIndex + 1], label);
-    return ArrayUtils.add(backPropagateFromTo(lastLayerIndex - 1, end, activations, error), error);
+    final INDArray error = layers[lastLayerIndex].backPropagate(label, activations[lastLayerIndex + 1], EMPTY);
+
+    if (lastLayerIndex == end) {
```

Comment:
This is the case that a neural network model has two layers and the last one is the `ActivationWithLossLayer`. I missed the case where the network model has only one layer.
In this case, the model cannot be trained since the gradients used for updating the layer should be calculated in the next layer's backpropagation. But, the next layer does not exist in the network.
I'll add a code for handling the case where the model has only one layer and comments about this.